### ISSUE TITLE
Wrong query parameter for `id` in strapi uses mongodb

### ISSUE THREAD

**Issue by alirezakay** [strapi/strapi#5667]
## Describe the bug

When I want to change the **permissions** of a role, I confront an unexpected error so that it couldn't be updated: ***'entry.notFound'***

![error](https://user-images.githubusercontent.com/20626510/78069708-caf8de80-73af-11ea-8ac3-b8398a7d1230.png)

---

I **traced** the logs and found out where the error originated in (*"node_modules/strapi-connector-mongoose/lib/queries.js"*):

![traced](https://user-images.githubusercontent.com/20626510/78070064-68eca900-73b0-11ea-8219-198ea2aad594.png)

So I printed the value of `params` (the input parameter of the function mentioned above):
```
{ id: "5e8338177069c5230a321064" }
```
and this value was being passed into the `findOne` function of MongoDB model.
Now it's obvious why this error occurred. It's because of the lack of this document in the MongoDB `users-permissions_role` collection:

```
{
        "_id" : ObjectId("5e8338177069c5230a321064"),
        "name" : "Authenticated",
        "description" : "Default role given to authenticated user.",
         ...
}
```

**The problem is**:

- the parameter key must be `_id` not `id` 
- the parameter value must be `ObjectId("5e8338177069c5230a321064")` not `"5e8338177069c5230a321064"`

because of these problems, the function `findOne` returns a null value and the error blows.

Please fix this issue :)

## Steps to reproduce the behaviour

1. create a **quick-start** strapi project.
2. create some content types and APIs in **development mode**.
3. think about having changed the database from **SQLite3** to **MongoDB** in **production mode**.
4. change the **database configs** in *./config/environments/production/database.json* from the root of the project directory
5. start the strapi project in production mode and see all the functionalities work very well with MongoDB (you need to configure some stuffs for using MongoDB though)
6. go to the **Roles and Permissions** tab of the admin panel and try to update permission and click save button (check/uncheck one of the action boxes).
7. duh, see nothing happens.
8. open the console of your browser and see there was an error in the API request.
9. check the terminal that the strapi server is running on.
10. now see the low-detailed error :(

## Expected behaviour

Updating the permission for a role.

## System

- Node.js Version: 12.16.1
- NPM version: 6.13.4
- Strapi version: v3.0.0-beta.19.4
- Database: Mongodb v4.2.5
- Operating system: Ubuntu 16.04

**Comment by derrickmehaffy** [comment]
Yeah `id` is hard coded and all references to `_id` should be mapped to `id` for mongoose. This needs to be changed if there is any collections that **don't** have an alias field of `id`

**Comment by lauriejim** [comment]
Hello! I just tried to reproduce your issue by following steps but I was not able to reproduce.
Here is what I did - https://www.loom.com/share/73eb7df0d41547188a99d5263d7578f8
Did I miss something?

**Comment by rmaroun** [comment]
This issue could also be related: #5751

**Comment by aksdevac** [comment]
Hi, checked this because found it interesting, there is function named`replaceIdByPrimaryKey` in `starpi-database` which replaces id with model's primary key(_id) so this should not happen, not able to regenerate on beta-19.5. but I checked on existing project(for both dev, prod env) with MongoDB selected from start.

**Comment by alirezakay** [comment]
> Hello! I just tried to reproduce your issue by following steps but I was not able to reproduce.
> Here is what I did - https://www.loom.com/share/73eb7df0d41547188a99d5263d7578f8
> Did I miss something?

Yes, I guess so.
You did the first steps very well as I stated before but, you should build and run the project in production mode:

```bash
NODE_ENV=production yarn build
NODE_ENV=production yarn start
```

**Comment by llavre** [comment]
Got same issue in my project after updating it from 19.3 to 19.5.

**Comment by phil-warner** [comment]
I'm getting exactly the same issue as @alirezakay when trying to update permissions for a role. The update method in queries.js can't find the document using the params it receives.

I can reproduce it on the back end by opening the mongo shell and modifying the params as follows:

```
db.getCollection('users-permissions_role').find({ id: '5e8e24c94b1fcb0ce07e1b9a' })
// original params - returns nothing

db.getCollection('users-permissions_role').find({ _id: ObjectId('5e8e24c94b1fcb0ce07e1b9a') })
// returns the relevant document
{ "_id" : ObjectId("5e8e24c94b1fcb0ce07e1b9a"), "name" : "Authenticated", "description" : "Default role given to authenticated user.", "type" : "authenticated", "createdAt" : ISODate("2020-04-08T19:23:53.337Z"), "updatedAt" : ISODate("2020-04-08T19:23:53.337Z"), "__v" : 0 }
```

**Comment by lauriejim** [comment]
Hello! I tried again @alirezakay and it works well.
Here is what I did - https://www.loom.com/share/22ca378e05ad402881e0d4e77d54fcd5
I used your commands.
